storage/Action.cc

```diff
@@ -88,20 +87,23 @@ namespace storage
 		{
 		    // children of parents must be deleted beforehand
 
-		    Devicegraph::Impl::vertex_descriptor q = actiongraph.get_devicegraph(LHS)->get_impl().find_vertex(parent_sid);
-
-		    Devicegraph::Impl::adjacency_iterator vi2, vi2_end;
-		    for (boost::tie(vi2, vi2_end) = adjacent_vertices(q, actiongraph.get_devicegraph(LHS)->get_impl().graph); vi2 != vi2_end; ++vi2)
+		    if (!get_device(actiongraph)->get_impl().has_dependency_manager())
 		    {
-			sid_t child_sid = actiongraph.get_devicegraph(LHS)->get_impl()[*vi2]->get_sid();
+			Devicegraph::Impl::vertex_descriptor q = actiongraph.get_devicegraph(LHS)->get_impl().find_vertex(parent_sid);
 
-			vector<Actiongraph::Impl::vertex_descriptor> tmp = actiongraph.actions_with_sid(child_sid, ONLY_LAST);
-			if (!tmp.empty())
+			Devicegraph::Impl::adjacency_iterator vi2, vi2_end;
+			for (boost::tie(vi2, vi2_end) = adjacent_vertices(q, actiongraph.get_devicegraph(LHS)->get_impl().graph); vi2 != vi2_end; ++vi2)
 			{
-			    // Make sure it's a delete action
-			    const Action::Base* tmp_action = actiongraph[tmp.front()];
-			    if (dynamic_cast<const Action::Delete*>(tmp_action))
-				actiongraph.add_edge(tmp.front(), vertex);
+			    sid_t child_sid = actiongraph.get_devicegraph(LHS)->get_impl()[*vi2]->get_sid();
+
+			    vector<Actiongraph::Impl::vertex_descriptor> tmp = actiongraph.actions_with_sid(child_sid, ONLY_LAST);
+			    if (!tmp.empty())
+			    {
+				// Make sure it's a delete action
+				const Action::Base* tmp_action = actiongraph[tmp.front()];
+				if (dynamic_cast<const Action::Delete*>(tmp_action))
```

Comment:
If this is used more often, this might be a good candidate for a one-line inline function isDeleteAction().